Source/WebGPU/WebGPU/RenderPassEncoder.mm

```diff
@@ -118,8 +143,14 @@
     if (m_vertexDynamicOffsets.size())
         [m_renderCommandEncoder setVertexBytes:&m_vertexDynamicOffsets[0] length:m_vertexDynamicOffsets.size() * sizeof(m_vertexDynamicOffsets[0]) atIndex:m_device->maxBuffersPlusVertexBuffersForVertexStage()];
 
-    if (m_fragmentDynamicOffsets.size())
-        [m_renderCommandEncoder setFragmentBytes:&m_fragmentDynamicOffsets[0] length:m_fragmentDynamicOffsets.size() * sizeof(m_fragmentDynamicOffsets[0]) atIndex:m_device->maxBuffersForFragmentStage()];
+    ASSERT(m_fragmentDynamicOffsets.size());
+    static_assert(sizeof(m_fragmentDynamicOffsets[0]) == sizeof(m_minDepth), "expect dynamic offsets container to have matching size to depth values");
+    auto fragmentsLastBufferSize = m_fragmentDynamicOffsets.size();
+    ASSERT(fragmentsLastBufferSize >= fragmentValuesAfterDynamicOffsets);
+    using destType = std::remove_reference<decltype(m_fragmentDynamicOffsets[0])>::type;
+    m_fragmentDynamicOffsets[fragmentsLastBufferSize - fragmentValuesAfterDynamicOffsets] = as_type<destType>(m_minDepth);
```

Comment:
Would WTF::bitwise_cast work here?

Reply:
Oh nice, I'll use that. I was unaware of it, thank you.